spec/docker/container_spec.rb

```diff
@@ -84,22 +84,40 @@
   end
 
   describe '#stats', docker_1_9: true do
-    subject {
-      described_class.create('Cmd' => "echo hello", 'Image' => 'debian:wheezy')
-    }
-    after(:each) { subject.remove }
+    after(:each) do
+      subject.kill!
+      subject.remove
+    end
 
     context "when requesting container stats" do
+      subject {
+        described_class.create('Cmd' => ['echo', 'hello'], 'Image' => 'debian:wheezy')
+      }
+
       let(:output) { subject.stats }
       it "returns a Hash" do
         expect(output).to be_a Hash
       end
     end
+
+    context "when streaming container stats" do
+      subject {
+        described_class.create('Cmd' => ['sleep', '1'], 'Image' => 'debian:wheezy')
+      }
+
+      it "yields a Hash" do
+        subject.start! # If the container isn't started, no stats will be streamed
+        subject.stats do |output|
+          expect(output).to be_a Hash
```

Comment:
Done. 

NB: The API is kinda funky as if the container stop, the connection isn't closed, and we keep trying to read it forever. Not sure what we could do about it from the client side.